### DISCUSSION TITLE
Power Supply Options

### DISCUSSION THREAD

**Discussion by CGMunich** [MiczFlor/RPi-Jukebox-RFID#1745] [Ideas]
Hi, first of all thanks to everybody for making this great project possible!
I build a Phoniebox for my son and he loves it, so does my wife.

I wanted to share how I implemented the power supply since that is a bit tricky in my opinion as I did not find any solution I like.
I did not want to use a Powerbank because of several details that make it complicated to use (shuts down when empty without shutting down the raspberry pi properly, power fluctuations when charging a power bank, etc. ...)
Instead I was looking for some kind of accessory for the pi, that adds a battery and charging to it.
But all I found was cheap LiPo Batteries and some charging circuit that I did not trust.

But finally I found 2 nice options I wanted to share here. (BTW: I am not affiliated to the companies making them)
First I found a Hat / Module for the Raspberry that can switch between 3 power sources (3.5-28V or so), has a build in stepUp/Down converter to provide 5V and has a push button to boot and power down the pi. It is called:  **Zero2Go Omini** 
I use two of the inputs, one for a 9V power supply and the other one is connected to a battery box that can hold 4 AA Batteries or rechargeable NiMH Batteries. The batteries can provide power for 4 hours or so. I also modifies the Phoniebox Software, so the battery voltage level is displayed and there is a automatic shutdown of the system when the batteries are below a certain level ( I use 1V per NiMH).  Battery power is cut after power down so the battery is not drained by Standby current draw,
Only downside is the there is no charging implemented. This setup is used for about 1.5 years already.

The other option I found and working on integrating now is the **StromPi** with a 1000mAh LiFePO4 Battery. (this battery type is safer and more robust than LiPo). This accessory also has several power inputs and the battery voltage can also be read from the Phoniebox Software with a little modification. It also powers up the Pi with a push-button but the System Shutdown must be done initiated by Software ( i.e the Phoniebox timeout timers). When shutting down, a provided script initiates Linux to shut down and then shuts down the StromPi. If the sudo shutdown is used the Strom-Pi cannot wake up the Pi again.
Also power is cut completely to save battery.

The enclosure I used is from Ikea, but not sold anymore. The speakers came with a Raspberry Pi Sound Interface and fit exactly in the box.

I hope this is useful for somebody! Let me know if you have questions.
Use this information at your own risk, as I might have gotten something wrong and cannot guarantee anything.

![IMG_6649](https://user-images.githubusercontent.com/86932146/151864804-8fc388f6-69c0-497c-8bd8-0d21031e1afa.jpeg)

![IMG_6647](https://user-images.githubusercontent.com/86932146/151864822-fdb3844a-6c4e-4678-93fe-0ea725f6061b.jpeg)

![IMG_6648](https://user-images.githubusercontent.com/86932146/151864810-3455d671-4648-4963-a74a-c3a2a648f4b0.jpeg)

**Comment by pabera** [comment]
Very cool btw!
